climpred/bootstrap.py

```diff
@@ -498,87 +498,133 @@ def bootstrap_compute(
+                and metric.probabilistic
+                and 'init' in init_skill.coords
+            ):
+                init_skill['init'] = hind.init.values
+            bootstrapped_init_skill.append(init_skill)
+            # generate uninitialized ensemble from hist
+            uninit_hind = resample_uninit(hind, hist)
+            # compute uninit skill
+            bootstrapped_uninit_skill.append(
+                compute(
+                    uninit_hind,
                     verif,
+                    alignment=alignment,
                     metric=metric,
-                    alignment=reference_alignment,
+                    comparison=comparison,
+                    dim=dim,
                     add_attrs=False,
                     **metric_kwargs,
                 )
             )
-    init = xr.concat(init, dim='iteration', **CONCAT_KWARGS)
-    uninit = xr.concat(uninit, dim='iteration', **CONCAT_KWARGS)
-    # when persistence is not computed set flag
-    if pers != []:
-        pers = xr.concat(pers, dim='iteration', **CONCAT_KWARGS)
-        pers_output = True
-    else:
-        pers_output = False
+            # compute persistence skill
+            # impossible for probabilistic
+            if not metric.probabilistic:
+                pers_skill.append(
+                    reference_compute(
+                        smp_hind,
+                        verif,
+                        metric=metric,
+                        alignment=reference_alignment,
+                        add_attrs=False,
+                        **metric_kwargs,
+                    )
+                )
+        bootstrapped_init_skill = xr.concat(
+            bootstrapped_init_skill, dim='iteration', **CONCAT_KWARGS
+        )
+        bootstrapped_uninit_skill = xr.concat(
+            bootstrapped_uninit_skill, dim='iteration', **CONCAT_KWARGS
+        )
+        if pers_skill != []:
+            bootstrapped_pers_skill = xr.concat(
+                pers_skill, dim='iteration', **CONCAT_KWARGS
+            )
+            pers_output = True
+        else:
+            pers_output = False
 
-    # get confidence intervals CI
-    init_ci = _distribution_to_ci(init, ci_low, ci_high)
-    uninit_ci = _distribution_to_ci(uninit, ci_low, ci_high)
-    # probabilistic metrics wont have persistence forecast
-    # therefore only get CI if persistence was computed
-    if pers_output:
-        if set(pers.coords) != set(init.coords):
-            init, pers = xr.broadcast(init, pers)
-        pers_ci = _distribution_to_ci(pers, ci_low_pers, ci_high_pers)
     else:
-        # otherwise set all persistence outputs to false
-        pers = init.isnull()
-        pers_ci = init_ci == -999
+        bootstrapped_hind = _resample_iterations_idx(hind, iterations, resample_dim)
+        # uninit
+        # create more members than needed
+        uninit_hind = xr.concat(
+            [resample_uninit(hind, hist) for i in range(4)], dim='member'
```

Comment:
I think we could eventually refactor this function even to be a little bit quicker with our new indexing method? Instead of looping through as we do there currently. (I'm thinking for the hindcast form of this)

Reply:
I apply theresampling afterwards. I first want to create uninit members. I want to avoid Resampling 100x just from 10 members. With the for loop I create here 4x10 members t reaample from. Is this now more clear why I want to do that?